modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/JSON/TokenGroupIdentifier.swift

```diff
@@ -130,11 +130,13 @@ public protocol TokenGroupIdentifierProtocol {
     static func identifier(fromFileName: String) -> TokenGroupIdentifierProtocol?
     func identify(token: TokenGroupIdentifiable) -> TokenGroup
     func hasContract(address: String, chainID: Int) -> Bool
+    func isSpam(address: String, chainID: Int) -> Bool
 }
 
 public class TokenGroupIdentifier: TokenGroupIdentifierProtocol {
 
     private var decodedTokenEntries: TokenGroupDictionary = TokenGroupDictionary()
+    private var spamTokenEntries: Set<String> = Set<String>()
```

Comment:
Both `var` can be `let` right?

Reply:
```swift
    private init(decodedTokenEntries: TokenGroupDictionary) {
        decodedTokenEntries.forEach { key, value in
            if value == TokenGroup.spam {
                spamTokenEntries.insert(key.lowercased())
            } else {
                self.decodedTokenEntries[key] = value
            }
        }
    }
```
Not without changing this code. Do you want me to?

Reply:
Yeah, but you can do it with a follow up PR to keep this PR smaller if you like. (`skip-review` if you are confident)